spec/syntax.md

```diff
@@ -87,285 +87,253 @@ The syntax specification takes into account the following design restrictions:
+> in the structure of a _message_. While many of the examples in this spec are written
+> on multiple lines, the formatting shown in this spec is primarily for readability.
+>
+>> Example.
+>> This _message_:
+>> ```
+>> let $foo = {|horse|}
+>> {You have a {$foo}!}
+>>```
+>> 
+>> Can also be written as:
+>> ```
+>> let $foo={|horse|}{You have a {$foo}!}
+>> ```
+>
+> An exception to this is: whitespace inside a _pattern_ is _always_ significant.
 
-A _selector_ selects a specific _pattern_ from a list of available _patterns_
-in a _message_ based on the value of its _expression_.
-A message can have multiple selectors.
+A _message_ consists of two parts:
+1. an optional list of _declarations_, followed by
+2. a _body_
 
-A message with a single _selector_:
+All _messages_ MUST contain a _body_. 
+An empty string is not a valid _message_.
 
-    match {$count :number}
-    when 1 {You have one notification.}
-    when * {You have {$count} notifications.}
+> A simple message:
+>```
+>{Hello, world!}
+>```
+>The same message defined in a `.properties` file:
+>
+>```properties
+>app.greetings.hello = {Hello, world!}
+>```
+>
+>The same message defined inline in JavaScript:
+>
+>```js
+>let hello = new MessageFormat('{Hello, world!}')
+>hello.format()
+>```
 
-A message with a single _selector_ which is an invocation of
-a custom function `:platform`, formatted on a single line:
+A _message_ satisfying all rules of the grammar is considered _well-formed_.
 
-    match {:platform} when windows {Settings} when * {Preferences}
+Furthermore, a _well-formed_ _message_ is considered _valid_
+if it meets additional semantic requirements about its structure, defined below.
 
-A message with a single _selector_ and a custom `:hasCase` function
-which allows the message to query for presence of grammatical cases required for each variant:
+### Declarations
 
-    match {$userName :hasCase}
-    when vocative {Hello, {$userName :person case=vocative}!}
-    when accusative {Please welcome {$userName :person case=accusative}!}
-    when * {Hello!}
+A **_declaration_** binds a variable identifier
+to the value of an _expression_ within the scope of a _message_.
+This local variable can then be used in other _expressions_ within the same _message_.
 
-A message with 2 _selectors_:
+```abnf
+declaration = let s variable [s] "=" [s] expression
+```
 
-    match {$photoCount :number} {$userGender :equals}
-    when 1 masculine {{$userName} added a new photo to his album.}
-    when 1 feminine {{$userName} added a new photo to her album.}
-    when 1 * {{$userName} added a new photo to their album.}
-    when * masculine {{$userName} added {$photoCount} photos to his album.}
-    when * feminine {{$userName} added {$photoCount} photos to her album.}
-    when * * {{$userName} added {$photoCount} photos to their album.}
+### Body
 
-### Local Variables
+The **_body_** of a message consists of either a _pattern_ or a _selector_.
```

Comment:
In line with my other comment about the `match` statement containing selectors and variant, I'd phrase this as:

```suggestion
The **_body_** of a message consists of either a _pattern_ or a _match statement_.
```